src/serde.rs

```diff
@@ -0,0 +1,92 @@
+//! CID Serde (de)serialization for the IPLD Data Model.
+//!
+//! CIDs cannot directly be represented in any of the native Serde Data model types. In order to
+//! work around that limitation. a newtype struct is introduced, that is used as a marker for Serde
+//! (de)serialization.
+use std::convert::TryFrom;
+use std::fmt;
+
+use serde::{de, ser};
+
+use crate::Cid;
+
+/// The newtype struct name that is used by Serde internally.
+pub const CID_SERDE_NEWTYPE_STRUCT_NAME: &str = "$__serde__newtype__struct__for__cid";
```

Comment:
Honestly, a simple "serde-cid" is probably enough to not conflict.

Reply:
Better me safe then sorry :)